run/jobs/src/test/java/com/example/JobsIntegrationTests.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example;
+
+import static org.junit.Assert.assertTrue;
+
+import com.google.api.gax.paging.Page;
+import com.google.cloud.logging.LogEntry;
+import com.google.cloud.logging.Logging;
+import com.google.cloud.logging.Logging.EntryListOption;
+import com.google.cloud.logging.LoggingOptions;
+import com.google.cloud.logging.Payload;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.channels.InterruptedByTimeoutException;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.TimeZone;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class JobsIntegrationTests {
+
+  private static final String project = System.getenv("GOOGLE_CLOUD_PROJECT");
+  private static final String suffix = UUID.randomUUID().toString();
+  private static String service;
+
+  @BeforeClass
+  public static void setup() throws Exception {
+    service = "job-quickstart-f374953a-498c-43b8-847a-7998ae5665d5"; // "job-quickstart-" + suffix;
+
+    ProcessBuilder setup = new ProcessBuilder();
+    setup.command(
+        "gcloud",
+        "builds",
+        "submit",
+        "--project=" + project,
+        "--config=./src/test/java/com/example/resources/e2e_test_setup.yaml",
+        String.format("--substitutions=_SERVICE=%s,_VERSION=%s", service, suffix));
```

Comment:
I always wonder why we do this?  Why we think it's a good idea?

Reply:
What specifically? Using substitutions - I want my job and image to have UUID tags, using String.format - looks nicer than +, Cloud Build - no "easy" programmatic way to build a container or deploy to Cloud Run.

Reply:
Running commands from code in our tests.